src/litserve/loops/base.py

```diff
@@ -250,23 +250,19 @@ def populate_context(self, lit_spec: LitSpec, request: Any):
     def put_response(
         self, response_queues: List[Queue], response_queue_id: int, uid: str, response_data: Any, status: LitAPIStatus
     ) -> None:
-        if self.socket:
-            self.socket.send_pyobj((uid, (response_data, status)))
+        if self.producer:
+            self.producer.put((uid, (response_data, status)), consumer_id=response_queue_id)
         else:
             response_queues[response_queue_id].put((uid, (response_data, status)), block=False)
 
     def put_error_response(
         self, response_queues: List[Queue], response_queue_id: int, uid: str, error: Exception
     ) -> None:
-        if self.socket:
-            self.socket.send_pyobj((uid, (error, LitAPIStatus.ERROR)))
-        else:
-            response_queues[response_queue_id].put((uid, (error, LitAPIStatus.ERROR)), block=False)
+        self.put_response(response_queues, response_queue_id, uid, error, LitAPIStatus.ERROR)
 
     def __del__(self):
-        if self.socket:
-            self.socket.close(linger=0)
-            self.zmq_ctx.term()
+        if self.producer:
+            self.producer.close()
```

Comment:
this isn't guaranteed to be called. maybe register an exit signal instead?

Reply:
updated